graph/src/blockchain/firehose_block_ingestor.rs

```diff
@@ -97,6 +178,99 @@ where
+            ExponentialBackoff::new(Duration::from_millis(250), Duration::from_secs(30));
+        loop {
+            match self.chain_store.clone().chain_backfill_cursor() {
+                Ok(cursor) => return cursor.unwrap_or_else(|| "".to_string()),
+                Err(e) => {
+                    error!(
+                        self.logger,
+                        "Fetching chain backfill cursor failed: {:?}", e
+                    );
+
+                    backoff.sleep_async().await;
+                }
+            }
+        }
+    }
+
+    async fn fetch_backfill_target_block_num(&self) -> BlockNumber {
+        let mut backoff =
+            ExponentialBackoff::new(Duration::from_millis(250), Duration::from_secs(30));
+        loop {
+            match self.chain_store.clone().chain_backfill_target_block_num() {
+                Ok(opt) => {
+                    return match opt {
+                        None => {
+                            // no target block number set yet. will determine and save initial target block from raw blocks table
+                            match self.determine_backfill_target_block_num().await {
+                                None => {
+                                    info!(self.logger, "Could not yet determine backfill target block number: no blocks set yet for this chain");
+                                    backoff.sleep_async().await;
+                                    continue;
+                                }
+                                Some(block_num) => {
+                                    self.set_backfill_target_block_num(block_num).await;
+                                    block_num
+                                }
+                            }
+                        }
+                        Some(block_num) => block_num.into(),
+                    };
+                }
+                Err(e) => {
+                    error!(
+                        self.logger,
+                        "Fetching chain backfill target failed: {:?}", e
+                    );
+                    backoff.sleep_async().await;
+                }
+            }
+        }
+    }
+
+    async fn determine_backfill_target_block_num(&self) -> Option<BlockNumber> {
+        let mut backoff =
+            ExponentialBackoff::new(Duration::from_millis(250), Duration::from_secs(30));
+
+        loop {
+            match self.chain_store.chain_initial_backfill_target_block_num() {
+                Ok(opt) => {
+                    return match opt {
+                        None => None,
+                        Some(t) => Some(t),
+                    }
+                }
+                Err(e) => {
+                    error!(self.logger, "Setting chain backfill target failed: {:?}", e);
+                    backoff.sleep_async().await;
+                }
+            }
+        }
+    }
+
+    async fn set_backfill_target_block_num(&self, block_num: BlockNumber) {
```

Comment:
When splitting `fetch_backfill_target_block_num`, I don't think this will be necessary and could be put straight in `initialize_backfill_target_block_num` which would make the retry portion clearer and located in the `initialize_backfill_target_block_num` directly.

Reply:
done!